static/src/javascripts/projects/common/modules/atoms/youtube.js

```diff
@@ -252,55 +263,84 @@ const STATES = {
     PAUSED: onPlayerPaused,
 };
 
-const getIFrameBehaviour = (atomId: string): IFrameBehaviour => {
-    const isAutoplayBlockingPlatform = () => isIOS() || isAndroid();
+const getIFrameBehaviourConfig = (
+    iframe: HTMLIFrameElement
+): IFrameBehaviourConfig => {
+    const isAutoplayBlockingPlatform = isIOS() || isAndroid();
 
-    const isInternalReferrer = (): boolean => {
+    const isInternalReferrer = ((): boolean => {
         if (config.get('page.isDev')) {
             return document.referrer.indexOf(window.location.origin) === 0;
         }
 
         return document.referrer.indexOf(config.get('page.host')) === 0;
-    };
+    })();
 
-    const isMainVideo = (): boolean =>
-        (players[atomId].iframe &&
-            !!players[atomId].iframe.closest(
-                'figure[data-component="main video"]'
-            )) ||
+    const isMainVideo =
+        (iframe && !!iframe.closest('figure[data-component="main video"]')) ||
         false;
 
-    const flashingElementsAllowed = (): boolean =>
-        accessibilityIsOn('flashing-elements');
+    const flashingElementsAllowed = accessibilityIsOn('flashing-elements');
 
-    const isVideoArticle = (): boolean =>
+    const isVideoArticle =
         config.get('page.contentType', '').toLowerCase() === 'video';
 
-    const isFront = () => config.get('page.isFront');
+    const isFront = config.get('page.isFront');
     const isUSContent =
         config.get('page.productionOffice', '').toLowerCase() === 'us';
 
     const isPaidContent = config.get('page.isPaidContent');
+
+    return {
+        isAutoplayBlockingPlatform,
+        isInternalReferrer,
+        isMainVideo,
+        flashingElementsAllowed,
+        isVideoArticle,
+        isFront,
+        isUSContent,
+        isPaidContent,
+    };
+};
+
+const getIFrameBehaviour = (
+    iframeConfig: IFrameBehaviourConfig
+): IFrameBehaviour => {
+    let iFrameBehaviour;
```

Comment:
Minor: I have a preference for avoiding `let` as much as possible. Do we need to declare a mutable variable here? Could we return the object instead:

```js
if (isUsPaidContentVideo) {
    return {
        autoplay: isUsPaidContentVideo,
        ...
    }
} else return {
        ...
```

Reply:
Sure, happy  to change that. I'd always been told that having a single exit point from a function is preferable as it makes it easier to know where you're leaving that function when you need to debug, but also happy to learn new styles and approaches! Will swap it to using  the 
```javascript
if (isUsPaidContentVideo) {
    return {
        autoplay: isUsPaidContentVideo,
        ...
    }

return {
    autoplay: 
    ...
}
```

suggested below.